test/plugin/test_out_google_cloud_grpc.rb

```diff
@@ -0,0 +1,280 @@
+        # gets propagated through the plugin
+        d.emit('message' => log_entry(0))
+        begin
+          d.run
+        rescue GRPC::Cancelled => error
+          assert_equal "GRPC::#{message}", error.message
+          exception_count += 1
+        rescue GRPC::BadStatus => error
+          assert_equal "#{code}:#{message}", error.message
+          exception_count += 1
+        end
+      end
+      assert_equal 1, @failed_attempts.size, "Index #{index} failed."
+      assert_equal 1, exception_count, "Index #{index} failed."
+    end
+  end
+
+  private
+
+  GRPC_MOCK_HOST = 'localhost:56789'
+
+  WriteLogEntriesRequest = Google::Logging::V1::WriteLogEntriesRequest
+  WriteLogEntriesResponse = Google::Logging::V1::WriteLogEntriesResponse
+
+  USE_GRPC_CONFIG = %(
+    use_grpc true
+  )
+
+  HTTP_REQUEST_MESSAGE = {
+    'requestMethod' => 'POST',
+    'requestUrl' => 'http://example/',
+    'requestSize' => 210,
+    'status' => 200,
+    'responseSize' => 65,
+    'userAgent' => 'USER AGENT 1.0',
+    'remoteIp' => '55.55.55.55',
+    'referer' => 'http://referer/',
+    'cacheHit' => true,
+    'cacheValidatedWithOriginServer' => true
+  }
+
+  HTTP_REQUEST_MESSAGE_WITHOUT_REFERER = {
+    'requestMethod' => 'POST',
+    'requestUrl' => 'http://example/',
+    'requestSize' => 210,
+    'status' => 200,
+    'responseSize' => 65,
+    'userAgent' => 'USER AGENT 1.0',
+    'remoteIp' => '55.55.55.55',
+    'cacheHit' => true,
+    'cacheValidatedWithOriginServer' => true
+  }
+
+  def create_driver(conf = APPLICATION_DEFAULT_CONFIG, tag = 'test',
+                    grpc_stub = GRPCLoggingMockService.rpc_stub_class)
+    conf += USE_GRPC_CONFIG
+    Fluent::Test::BufferedOutputTestDriver.new(
+      GoogleCloudOutputWithGRPCMock.new(grpc_stub), tag).configure(conf, true)
+  end
+
+  # Google Cloud Fluent output stub with grpc mock.
+  class GoogleCloudOutputWithGRPCMock < Fluent::GoogleCloudOutput
+    def initialize(grpc_stub)
+      super()
+      @grpc_stub = grpc_stub
+    end
+
+    def api_client
+      ssl_creds = GRPC::Core::ChannelCredentials.new
+      authentication = Google::Auth.get_application_default
+      creds = GRPC::Core::CallCredentials.new(authentication.updater_proc)
+      ssl_creds.compose(creds)
+
+      # Here we have obtained the creds, but for the mock, we will leave the
+      # channel insecure.
+      @grpc_stub.new(GRPC_MOCK_HOST, :this_channel_is_insecure)
+    end
+  end
+
+  # GRPC logging mock that successfully logs the records.
+  class GRPCLoggingMockService < Google::Logging::V1::LoggingService::Service
+    def initialize(requests_received)
+      super()
+      @requests_received = requests_received
+    end
+
+    def write_log_entries(request, _call)
+      @requests_received << request
+      WriteLogEntriesResponse.new
+    end
+
+    # TODO(lingshi) Remove these dummy methods when grpc/9033 is fixed.
+    def list_logs(_request, _call)
```

Comment:
Changed it to use `alias_method`. Note that the `alias_method` could not detect the `_undefined` method from the included `BastTest` module unless it's explicitly defined in the `GRPCLoggingMockService` class.